database/download.go

```diff
@@ -50,25 +60,33 @@ func (db *DB) DownloadByID(ctx context.Context, id primitive.ObjectID) (*Downloa
 
 // DownloadCreate registers a new download. Marks partial downloads by supplying
 // the `bytes` param. If `bytes` is 0 we assume a full download.
-func (db *DB) DownloadCreate(ctx context.Context, user User, skylink Skylink, bytes int64) (*Download, error) {
+func (db *DB) DownloadCreate(ctx context.Context, user User, skylink Skylink, bytes int64) error {
 	if user.ID.IsZero() {
-		return nil, errors.New("invalid user")
+		return errors.New("invalid user")
 	}
 	if skylink.ID.IsZero() {
-		return nil, errors.New("invalid skylink")
+		return errors.New("invalid skylink")
+	}
+
+	// Check if there exists a download of this skylink by this user, updated
+	// within the DownloadUpdateWindow and keep updating that, if so.
+	down, err := db.DownloadRecent(ctx, skylink.ID)
+	if err == nil && down.Updated.Add(DownloadUpdateWindow).After(time.Now().UTC()) {
```

Comment:
I guess this is fine but you could add the update window to the query imo

Reply:
Good suggestion. While implementing this I also noticed that the `updated_at` was never actually updated, so that's fixed as well.